File: 图片轮播器－oc/图片轮播器－oc/ZDCollectionViewCell.m
//
//  ZDCollectionViewCell.m
//  图片轮播器－oc
//
//  Created by zz on 16/2/27.
//  Copyright © 2016年 zz. All rights reserved.
//

#import "ZDCollectionViewCell.h"

@implementation ZDCollectionViewCell{

    UIImageView *_pictureView;
}
-(instancetype)initWithFrame:(CGRect)frame{

    self = [super initWithFrame:frame];
    if (self) {
        _pictureView = [[UIImageView alloc]initWithFrame:self.bounds];//self.frame=(0,0,0,0)
        
        [self.contentView addSubview:_pictureView];
        
    }
    return self;
}
-(void)setUrl:(NSURL *)url{

    _url = url;
    NSData *data = [NSData dataWithContentsOfURL:url];
    _pictureView.image = [UIImage imageWithData:data];
    
}

@end

File: 图片轮播器swift/图片轮播器swift/ZDCollectionViewCell.swift
//
//  ZDCollectionViewCell.swift
//  图片轮播器－swift
//
//  Created by zz on 16/3/2.
//  Copyright © 2016年 zz. All rights reserved.
//

import UIKit

class ZDCollectionViewCell: UICollectionViewCell {

    var url: NSURL?{
        didSet{
        
        imageView.image = UIImage(data: NSData(contentsOfURL: url!)!)
        }
    }
    
    override init(frame: CGRect) {
    
        super.init(frame: frame)
        
        setupUI()
        
    }

    private var imageView: UIImageView = UIImageView()
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
}
//MARK: - 界面设置
extension ZDCollectionViewCell{

    private func setupUI(){
    
        backgroundColor = UIColor.whiteColor()
        contentView.addSubview(imageView)
        imageView.frame = self.bounds
    }
}
